editor/import/editor_scene_importer_gltf.cpp

```diff
@@ -229,9 +229,9 @@ Error EditorSceneImporterGLTF::_parse_scenes(GLTFState &state) {
 
 	ERR_FAIL_COND_V(!state.json.has("scenes"), ERR_FILE_CORRUPT);
 	const Array &scenes = state.json["scenes"];
-	ERR_FAIL_COND_V(!scenes.size(), ERR_FILE_CORRUPT);
-	for (int i = 0; i < 1; i++) { //only first scene is imported
-		const Dictionary &s = scenes[i];
+	if (scenes.size() && state.json.has("scene")) {
```

Comment:
You're skipping all of `scenes` now and only loading the main `scene`. What about all the other non-main scenes?

Reply:
According to the spec:

https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#scenes

> The glTF asset contains zero or more scenes, the set of visual objects to render. Scenes are defined in a scenes array. An additional property, scene (note singular), identifies which of the scenes in the array is to be displayed at load time.

> When scene is undefined, runtime is not required to render anything at load time.

The previous behaviour was to pick scene 1 and crash on empty scenes sizes.

The new behaviour is to ignore empty scenes sizes, use the scene defined to choose from the scenes.

If the scene is undefined, I'm not sure what to do. Perhaps use the old behaviour and pretend it is scene one? Or the current pull request code behaviour where everything is skipped as mentioned in the glTF2 spec.

Reply:
But the spec doesn't say that we should skip the rest, only that the `scene` one is meant to be used at load time. Do we only support "load time"? What's the purpose of the non-"load time" scenes in glTF?

Reply:
The problem this solves for glTF is:

> Implementation Note: This allows applications to use glTF assets as libraries of individual entities such as materials or meshes.

However, we don't have a ui to choose a particular scene.

Reply:
The previous behaviour was to pick a scene at random (it's always the first scene).

I interpret the load time scene as the one the author intended to display.

I interpret non-"load time" scenes to be ones that is used as libraries of things that aren't in the scene but can be used by a specific application to store non-"load time" scene nodes because of the technical note.

Reply:
An alternative interpretation is we iterate all the scenes and merge the scenes into one if the "scene" is not defined.

This should also be valid.

Reply:
No one asked for that feature and it's possible but not required in the spec...

Reply:
Then I think we should maybe ignore `scenes` and only check `scene` for now, and use that. If `scene` is undefined, we can loads `scenes[0]` as you suggest, and also issue a warning about it, as it means we're doing our own interpretation of the file but it's not a "good" glTF asset for use in a game engine.

Reply:
Please take a look at the new revision. Will do some testing on the gltf https://github.com/KhronosGroup/glTF-Asset-Generator/releases/tag/v0.6.1 while the build churns.

Reply:
Ran through the animated scene models and everything was expected.